esp-hal/src/interrupt/riscv.rs

```diff
@@ -347,26 +474,30 @@ mod vectored {
         }
     }
 
-    /// Get the interrupts configured for the core
+    /// Get the interrupts configured for the core at the given priority
+    /// matching the given status
     #[inline]
-    fn get_configured_interrupts(core: Cpu, mut status: u128) -> [u128; 16] {
+    fn get_configured_interrupts(
```

Comment:
This is a _much_ better way of doing this, nice! We can also apply to Xtensa too, I think there is a very similar block of code. Let's do that in a separate PR though (I'll make a note in the tracking issue).